### PR TITLE
Feature: App Router integration

### PR DESCRIPTION
**Pull request by ronnyhaase**
- Moves current app into module "staking"
- Moves generic stuff into module "common"
- Adds basic routing
- Remove circular dependency between app and client

### CODE REVIEW

**Comment by ronnyhaase** [comment]
@dankelleher Does commit [9df645c](https://github.com/sunrise-stake/app/pull/131/commits/9df645c7da5f412b7a16be9cabc44422b6674434) make sense to you?

**Review by s6thgehr — DISMISSED** [review]

File: packages/app/src/common/utils.ts [hunk 1]
```diff
@@ -81,3 +76,18 @@ export function debounce<F extends (...args: Parameters<F>) => ReturnType<F>>(
     timeout = setTimeout(() => func(...args), waitFor);
   };
 }
+
+export { MAX_NUM_PRECISION, toSol } from "@sunrisestake/client";
```
Comment (dankelleher) [inline packages/app/src/common/utils.ts:80]: nit:

Is the idea here that the rest of the app will not have a dependency on the client but just on this file? My preference would be to leave this line out, and just have any parts of the app that need it just import these directly from the client.

**Review by dankelleher — APPROVED** [review]
Looks good - approved pending green build

File: packages/app/src/common/utils.ts (hunk 1 above)
Comment (ronnyhaase) [reply packages/app/src/common/utils.ts:80]: Makes sense, will change it that way.

**Pull request merged**